fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/witness/WitnessPagerAdapter.java

```diff
@@ -0,0 +1,28 @@
+package com.github.dedis.popstellar.ui.detail.witness;
+
+import androidx.annotation.NonNull;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentActivity;
+import androidx.viewpager2.adapter.FragmentStateAdapter;
+
+public class WitnessPagerAdapter extends FragmentStateAdapter {
+
+  public WitnessPagerAdapter(@NonNull FragmentActivity fragmentActivity) {
+    super(fragmentActivity);
+  }
+
+  @NonNull
+  @Override
+  public Fragment createFragment(int position) {
+    if (position == 0) {
+      return new WitnessesFragment();
+    } else {
+      return new WitnessMessageFragment();
+    }
+  }
+
+  @Override
+  public int getItemCount() {
+    return 2; // We have 2 tabs
```

Comment:
create a variable?

Reply:
I am not sure that having a variable here is really useful since the getter signature literally explains this, but the question would be could we have more items in the future, even if we do just changing the value 2 here would be enough since we will use the getter everywhere else where we want to get the item count.

Reply:
I fully agree with Tesa. Plus if they want to have more than 2 tabs they will have to look at how the ItemCount function is implemented so to me it seems appropriate so

Reply:
all right, that seems good them